pkg/client/config/config.go

```diff
@@ -117,6 +118,22 @@ func loadConfig(context string) (*rest.Config, error) {
 
 	// If the recommended kubeconfig env variable is set, or there
 	// is no in-cluster config, try the default recommended locations.
+	//
+	// NOTE: For default config file locations, upstream only checks
+	// $HOME for the user's home directory, but we can also use
+	// os/user.HomeDir when $HOME is unset.
+	//
+	// TODO(jlanford): could this be done upstream?
+	if _, ok := os.LookupEnv("HOME"); !ok {
```

Comment:
does this actually work?  It looks like homedir is called from a static variable initializer -- https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/loader.go#L51

Reply:
Oh good catch. Yeah this won't work. It looks like changing `RecommendedHomeFile` will work though since that's what actually ends up getting used: https://github.com/kubernetes/client-go/blob/7ec8a74ae980246e9d1a5734f45bd44fde93a5c1/tools/clientcmd/loader.go#L153

I'll update to just set that directly instead of the HOME env var.

Reply:
is that thread-safe?

I'm worried that we may have to do a more invasive solution here, like customizing the result of (or not using at all) the NewDefaultBlahBlahBlah method.

Reply:
I _believe_ this is now thread-safe. Now, we no longer change any global variables. Instead we create an instance of `loadingRules` and then append the extra file derived from `u.HomeDir` to the precedence list.

I manually tested this and it still works as expected.

Suggestions on adding a unit test for this? I'm not opposed, but the use of `user.Current()` and `u.HomeDir` makes it a little awkward. I don't love the idea of creating temporary files and directories in the home directories of anyone that runs the unit tests.